language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/PullModuleCodeAction.java

```diff
@@ -45,71 +45,50 @@
 
     @Override
     public boolean isEnabled(LanguageServerContext serverContext) {
-        //TODO: disabled since offline is yet to be supported through project api
-        return false;
+        return true;
     }
 
     @Override
     public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
                                                     DiagBasedPositionDetails positionDetails,
                                                     CodeActionContext context) {
-        if (!(diagnostic.message().startsWith(UNRESOLVED_MODULE))) {
+        if (!DiagnosticErrorCode.MODULE_NOT_FOUND.diagnosticId().equals(diagnostic.diagnosticInfo().code())) {
             return Collections.emptyList();
         }
 
-        String diagnosticMessage = diagnostic.message();
-        String uri = context.fileUri();
-        CommandArgument uriArg = CommandArgument.from(CommandConstants.ARG_KEY_DOC_URI, uri);
+        Optional<String> moduleName = positionDetails.diagnosticProperty(MISSING_MODULE_NAME_INDEX);
+        if (moduleName.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        LSClientCapabilities clientCapabilities = context.languageServercontext().get(LSClientCapabilities.class);
+        if (!clientCapabilities.getInitializationOptions().isPullModuleSupported()) {
+            return Collections.emptyList();
+        }
+
+        CommandArgument uriArg = CommandArgument.from(CommandConstants.ARG_KEY_DOC_URI, context.fileUri());
         List<Diagnostic> diagnostics = new ArrayList<>();
 
-        Matcher matcher = CommandConstants.UNRESOLVED_MODULE_PATTERN.matcher(
-                diagnosticMessage.toLowerCase(Locale.ROOT)
-        );
-        if (matcher.find() && matcher.groupCount() > 0) {
-            List<Object> args = new ArrayList<>();
-            String pkgName = matcher.group(1).trim();
-            String version = getVersion(context, pkgName, matcher);
-            args.add(CommandArgument.from(CommandConstants.ARG_KEY_MODULE_NAME, pkgName + version));
-            args.add(uriArg);
-            String commandTitle = CommandConstants.PULL_MOD_TITLE;
+        List<Object> args = new ArrayList<>();
+        args.add(uriArg);
+        args.add(CommandArgument.from(CommandConstants.ARG_KEY_MODULE_NAME, moduleName.get()));
 
-            CodeAction action = new CodeAction(commandTitle);
-            action.setKind(CodeActionKind.QuickFix);
-            action.setCommand(new Command(commandTitle, PullModuleExecutor.COMMAND, args));
-            action.setDiagnostics(CodeActionUtil.toDiagnostics(diagnostics));
-            return Collections.singletonList(action);
-        }
-        return Collections.emptyList();
+        String commandTitle = CommandConstants.PULL_MOD_TITLE;
+        CodeAction action = new CodeAction(commandTitle);
+        action.setKind(CodeActionKind.QuickFix);
+        action.setCommand(new Command(commandTitle, "ballerina.packages.pull", args));
```

Comment:
We can reuse the PULL_MODULE_COMMAND const here

Reply:
Yes. Missed it. Done!